src/lxc/conf.c

```diff
@@ -3315,6 +3344,32 @@ int lxc_clear_limits(struct lxc_conf *c, const char *key)
 	return 0;
 }
 
+int lxc_clear_sysctls(struct lxc_conf *c, const char *key)
+{
+	struct lxc_list *it, *next;
+	bool all = false;
+	const char *k = NULL;
+
+	if (strcmp(key, "lxc.sysctl") == 0)
+		all = true;
+	else if (strncmp(key, "lxc.sysctl.", sizeof("lxc.sysctl.")-1) == 0)
+		k = key + sizeof("lxc.sysctl.")-1;
+	else
+		return -1;
+
+	lxc_list_for_each_safe(it, &c->sysctls, next) {
+		struct lxc_sysctl *elem = it->elem;
+		if (!all && strcmp(elem->key, k) != 0)
+			continue;
+		lxc_list_del(it);
+		free(elem->key);
+		free(elem->value);
+		free(elem);
+		free(it);
+	}
```

Comment:
I don't think we need to do that, because our goal is to clean up a specific configuration, and now, it has reached its goal.If we return an error, the upper layer will be puzzled.